client/src/pages/shop/utils.js

```diff
@@ -0,0 +1,17 @@
+export const ascendingPrice = () => {
+  const sortedItems = [...category];
+  sortedItems.sort((a, b) => a.price - b.price);
+  setCategory(sortedItems);
+};
+
+export const decendingPrice = () => {
+  const sortedItems = [...category];
+  sortedItems.sort((a, b) => b.price - a.price);
+  setCategory(sortedItems);
+};
+
+export const latest = () => {
+  const sortedItems = [...category];
+  sortedItems.sort((a, b) => new Date(b.Date) - new Date(a.Date));
+  setCategory(sortedItems);
+};
```

Comment:
You have to pass category and setCategory as parameters to these functions else they will give you the error of category, setCategory not defined.
```suggestion
export const ascendingPrice = (category, setCategory) => {
  const sortedItems = [...category];
  sortedItems.sort((a, b) => a.price - b.price);
  setCategory(sortedItems);
};

export const decendingPrice = (category, setCategory) => {
  const sortedItems = [...category];
  sortedItems.sort((a, b) => b.price - a.price);
  setCategory(sortedItems);
};

export const latest = (category, setCategory) => {
  const sortedItems = [...category];
  sortedItems.sort((a, b) => new Date(b.Date) - new Date(a.Date));
  setCategory(sortedItems);
};
```